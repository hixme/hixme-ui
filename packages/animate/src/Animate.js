import styled, { keyframes } from 'styled-components'
import {
  bounceOut,
  bounce,
  bounceIn,
  bounceInDown,
  bounceInLeft,
  bounceInRight,
  bounceInUp,
  bounceOutDown,
  bounceOutLeft,
  bounceOutRight,
  bounceOutUp,
  fadeIn,
  fadeInDown,
  fadeInDownBig,
  fadeInLeft,
  fadeInLeftBig,
  fadeInRight,
  fadeInRightBig,
  fadeInUp,
  fadeInUpBig,
  fadeOut,
  fadeOutDown,
  fadeOutDownBig,
  fadeOutLeft,
  fadeOutLeftBig,
  fadeOutRight,
  fadeOutRightBig,
  fadeOutUp,
  fadeOutUpBig,
  flash,
  flip,
  flipInX,
  flipInY,
  flipOutX,
  flipOutY,
  headShake,
  hinge,
  jello,
  lightSpeedIn,
  lightSpeedOut,
  pulse,
  rollIn,
  rollOut,
  rotateIn,
  rotateInDownLeft,
  rotateInDownRight,
  rotateInUpLeft,
  rotateInUpRight,
  rotateOut,
  rotateOutDownLeft,
  rotateOutDownRight,
  rotateOutUpLeft,
  rotateOutUpRight,
  rubberBand,
  shake,
  slideInDown,
  slideInLeft,
  slideInRight,
  slideInUp,
  slideOutDown,
  slideOutLeft,
  slideOutRight,
  slideOutUp,
  swing,
  tada,
  wobble,
  zoomIn,
  zoomInDown,
  zoomInLeft,
  zoomInRight,
  zoomInUp,
  zoomOut,
  zoomOutDown,
  zoomOutLeft,
  zoomOutRight,
  zoomOutUp,
} from 'react-animations'

const chooseReactAnimation = (props) => {
  if (props.bounceOut) return bounceOut
  if (props.bounce) return bounce
  if (props.bounceIn) return bounceIn
  if (props.bounceInDown) return bounceInDown
  if (props.bounceInLeft) return bounceInLeft
  if (props.bounceInRight) return bounceInRight
  if (props.bounceInUp) return bounceInUp
  if (props.bounceOutDown) return bounceOutDown
  if (props.bounceOutLeft) return bounceOutLeft
  if (props.bounceOutRight) return bounceOutRight
  if (props.bounceOutUp) return bounceOutUp
  if (props.fadeIn) return fadeIn
  if (props.fadeInDown) return fadeInDown
  if (props.fadeInDownBig) return fadeInDownBig
  if (props.fadeInLeft) return fadeInLeft
  if (props.fadeInLeftBig) return fadeInLeftBig
  if (props.fadeInRight) return fadeInRight
  if (props.fadeInRightBig) return fadeInRightBig
  if (props.fadeInUp) return fadeInUp
  if (props.fadeInUpBig) return fadeInUpBig
  if (props.fadeOut) return fadeOut
  if (props.fadeOutDown) return fadeOutDown
  if (props.fadeOutDownBig) return fadeOutDownBig
  if (props.fadeOutLeft) return fadeOutLeft
  if (props.fadeOutLeftBig) return fadeOutLeftBig
  if (props.fadeOutRight) return fadeOutRight
  if (props.fadeOutRightBig) return fadeOutRightBig
  if (props.fadeOutUp) return fadeOutUp
  if (props.fadeOutUpBig) return fadeOutUpBig
  if (props.flash) return flash
  if (props.flip) return flip
  if (props.flipInX) return flipInX
  if (props.flipInY) return flipInY
  if (props.flipOutX) return flipOutX
  if (props.flipOutY) return flipOutY
  if (props.headShake) return headShake
  if (props.hinge) return hinge
  if (props.jello) return jello
  if (props.lightSpeedIn) return lightSpeedIn
  if (props.lightSpeedOut) return lightSpeedOut
  if (props.pulse) return pulse
  if (props.rollIn) return rollIn
  if (props.rollOut) return rollOut
  if (props.rotateIn) return rotateIn
  if (props.rotateInDownLeft) return rotateInDownLeft
  if (props.rotateInDownRight) return rotateInDownRight
  if (props.rotateInUpLeft) return rotateInUpLeft
  if (props.rotateInUpRight) return rotateInUpRight
  if (props.rotateOut) return rotateOut
  if (props.rotateOutDownLeft) return rotateOutDownLeft
  if (props.rotateOutDownRight) return rotateOutDownRight
  if (props.rotateOutUpLeft) return rotateOutUpLeft
  if (props.rotateOutUpRight) return rotateOutUpRight
  if (props.rubberBand) return rubberBand
  if (props.shake) return shake
  if (props.slideInDown) return slideInDown
  if (props.slideInLeft) return slideInLeft
  if (props.slideInRight) return slideInRight
  if (props.slideInUp) return slideInUp
  if (props.slideOutDown) return slideOutDown
  if (props.slideOutLeft) return slideOutLeft
  if (props.slideOutRight) return slideOutRight
  if (props.slideOutUp) return slideOutUp
  if (props.swing) return swing
  if (props.tada) return tada
  if (props.wobble) return wobble
  if (props.zoomIn) return zoomIn
  if (props.zoomInDown) return zoomInDown
  if (props.zoomInLeft) return zoomInLeft
  if (props.zoomInRight) return zoomInRight
  if (props.zoomInUp) return zoomInUp
  if (props.zoomOut) return zoomOut
  if (props.zoomOutDown) return zoomOutDown
  if (props.zoomOutLeft) return zoomOutLeft
  if (props.zoomOutRight) return zoomOutRight
  if (props.zoomOutUp) return zoomOutUp
  if (props.flipInX) return flipInX
  return bounce
}

const animationStyle = props => keyframes`${chooseReactAnimation(props)}`

const Animate = styled.div`
  animation: ${(props) => {
    if (props.time) return props.time
    return '1s'
  }} ${animationStyle};
  animation-delay: ${(props) => {
    if (props.delay) return props.delay
    return '0s'
  }};
  animation-iteration-count: ${(props) => {
    if (props.count) return props.count
    return '1'
  }};
  animation-play-state: ${(props) => {
    if (props.trigger) return 'running'
    return 'paused'
  }};
  animation-fill-mode: ${(props) => {
    if (props.fillMode) return props.fillMode
    return 'none'
  }};
  display: ${(props) => {
    if (props.display) return props.display
    return 'inline-block'
  }};
`

Animate.displayName = 'Animate'

export default Animate
