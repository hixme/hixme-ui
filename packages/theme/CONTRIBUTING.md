# Contributing

Open a PR at https://github.com/hixme/hixme-ui.
