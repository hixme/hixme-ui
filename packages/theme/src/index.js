export const colors = {
  white: 'white',
  black: 'black',
  grey30: '#3A4951',
  grey80: '#8F9DA8',
  grey100: '#AAB5BD',
  grey110: '#B9C5CD',
  grey120: '#CBD6DB',
  grey130: '#f7fcff',
  teal: '#2EAE8F',
  blue: '#3997E2',
  red: '#EB5565',
  yellow: '#E0BC4D',
  purple: '#A374D2',
  green: '#88b963',
}

const gradients = {
  teal: 'linear-gradient(180deg, #20BCA1 0%, #159777 100%)',
  red: 'linear-gradient(0deg, #DE293C 0%, #FF7272 100%)',
  yellow: 'linear-gradient(-1deg, #FF8C00 0%, #F3C200 100%)',
  blue: 'linear-gradient(-1deg, #2583C4 0%, #30A0EE 100%)',
  grey: 'linear-gradient(180deg, #CCD2D6 0%, #a4adb3 100%)',
}

const fontSizes = {
  jumbo: '22px',
  larger: '20px',
  large: '18px',
  medium: '16px',
  default: '14px',
  small: '12px',
  smaller: '10px',
  smallest: '8px',
}

const fontWeights = {
  default: 'normal',
}

const titleFontSizes = {
  jumbo: '38px',
  larger: '36px',
  large: '34px',
  medium: '32px',
  default: '30px',
  small: '28px',
  smaller: '26px',
  smallest: '24px',
}

const buttonFontSizes = {
  mini: fontSizes.small,
  small: fontSizes.default,
  large: fontSizes.large,
  jumbo: fontSizes.jumbo,
}

const buttonHeights = {
  mini: '28px',
  small: '36px',
  default: '42px',
  large: '50px',
  jumbo: '65px',
}

const buttonMinWidths = {
  mini: '80px',
  small: '80px',
  default: '100px',
  large: '100px',
  jumbo: '100px',
}

const buttonTextColors = {
  default: colors.white,
}

const buttonGradientColors = {
  primary: gradients.teal,
  secondary: gradients.grey,
  default: gradients.teal,
  red: gradients.red,
  danger: gradients.red,
  error: gradients.red,
  blue: gradients.blue,
  gold: gradients.yellow,
}

const shadows = {
  medium: '0px 5px 9px 0px rgba(56,59,60,0.22);',
  default: '0px 2px 6px 0px rgba(56,59,60,0.18);',
  none: '0px 0px 0px 0px rgba(0,0,0,0);',
}

const buttonBorderRadius = {
  none: '0',
  default: '4px',
}

/* eslint-disable max-len */
export const fontFamily = {
  thinnest: '\'Avenir-Next_UltraLight\', \'HelveticaNeue-UltraLight\', \'Helvetica Neue Ultra Light\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  thinner: '\'Avenir-Next_Light\', \'HelveticaNeue-Thin\', \'Helvetica Neue Thin\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  thin: '\'Avenir-Next_Regular\', \'HelveticaNeue-Light\', \'Helvetica Neue Light\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  normal: '\'Avenir-Next_Medium\', \'HelveticaNeue-Roman\', \'Helvetica Neue Roman\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  bold: '\'Avenir-Next_Demi\', \'HelveticaNeue-Medium\', \'Helvetica Neue Medium\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  bolder: '\'Avenir-Next_Bold\', \'HelveticaNeue-Bold\', \'Helvetica Neue Bold\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  boldest: '\'Avenir-Next_Heavy\', \'HelveticaNeue-Heavy\', \'Helvetica Neue Heavy\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
}

const defaults = {
  gradient: gradients.teal,
  fontSize: fontSizes.default,
  buttonHeight: buttonHeights.default,
  textColor: colors.grey30,
  fontFamily: fontFamily.normal,
  fontWeight: fontWeights.default,
  backgroundColor: colors.white,
  borderColor: colors.grey110,
  titleFontSize: titleFontSizes.default,
}

export const colorTheme = {
  default: colors.grey30,
  primary: colors.teal,
  success: colors.teal,
  warning: colors.yellow,
  error: colors.red,
  danger: colors.red,
  muted: colors.grey80,

  white: colors.white,
  black: colors.black,
  red: colors.red,
  blue: colors.blue,
  purple: colors.purple,
  yellow: colors.yellow,
  green: colors.green,

  lighter: colors.grey110,
  light: colors.grey80,
  lightest: colors.grey130,
}

const outlineButtonColors = {
  ...colorTheme,
  default: colors.grey110,
}

const theme = {
  colors: colorTheme,
  backgroundColors: {
    ...colorTheme,
    default: defaults.backgroundColor,
  },
  borderColors: {
    ...colorTheme,
    default: defaults.borderColor,
  },
  buttonTextColors,
  buttonGradientColors,
  buttonFontSizes,
  buttonHeights,
  buttonMinWidths,
  buttonBorderRadius,
  outlineButtonColors,
  defaults,
  fontSizes,
  fontFamily,
  fontWeights,
  gradients,
  shadows,
  textColors: {
    ...colorTheme,
    default: defaults.textColor,
  },
  titleFontSizes,
}

export default theme

export ThemeProvider from './ThemeProvider'
