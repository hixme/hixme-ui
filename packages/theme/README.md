# hixme-ui theme

```bash
npm i --save @hixme-ui/theme
```

## Theme

HUI theme colors and defaults

```javascript
import theme from '@hixme-ui/theme

const primary = theme.colors.primary
```


## ThemeProvider

HUI theme provider, which is just an implementation of styled-components
ThemeProvider with the HUI theme as the default

```javascript
import { ThemeProvider } from '@hixme-ui/theme'

const App = ({ children }) => (
  <ThemeProvider>
    {children}
  </ThemeProvider>
)
```
