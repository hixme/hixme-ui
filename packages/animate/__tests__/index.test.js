import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from '@hixme-ui/theme'
import Animate from '../src/index.js'

test('Animate', () => {
  const component = renderer.create(
    (<ThemeProvider><Animate /></ThemeProvider>),
  )

  const tree = component.toJSON()
  expect(tree.type).toBe('div')
})
