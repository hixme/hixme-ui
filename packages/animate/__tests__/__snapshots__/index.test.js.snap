// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animate Count Should have default count of 1 1`] = `
.c0 {
  -webkit-animation: hVlBnv;
  animation: hVlBnv;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-play-state: paused;
  animation-play-state: paused;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  display: inherit;
}

<div
  className="c0"
/>
`;

exports[`Animate Count Should take count prop 1`] = `
.c0 {
  -webkit-animation: hVlBnv;
  animation: hVlBnv;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: 2;
  animation-iteration-count: 2;
  -webkit-animation-play-state: paused;
  animation-play-state: paused;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  display: inherit;
}

<div
  className="c0"
/>
`;

exports[`Animate Delay Should have default display of inherit 1`] = `
.c0 {
  -webkit-animation: hVlBnv;
  animation: hVlBnv;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-play-state: paused;
  animation-play-state: paused;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  display: inherit;
}

<div
  className="c0"
/>
`;

exports[`Animate Delay Should take display prop 1`] = `
.c0 {
  -webkit-animation: hVlBnv;
  animation: hVlBnv;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-play-state: paused;
  animation-play-state: paused;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
  className="c0"
  display="inline-flex"
/>
`;

exports[`Animate Display Should have default display of inherit 1`] = `
.c0 {
  -webkit-animation: hVlBnv;
  animation: hVlBnv;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-play-state: paused;
  animation-play-state: paused;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  display: inherit;
}

<div
  className="c0"
/>
`;

exports[`Animate Display Should take display prop 1`] = `
.c0 {
  -webkit-animation: hVlBnv;
  animation: hVlBnv;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-play-state: paused;
  animation-play-state: paused;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
  className="c0"
  display="inline-flex"
/>
`;

exports[`Animate Fill Mode Should have default fill mode of none 1`] = `
.c0 {
  -webkit-animation: hVlBnv;
  animation: hVlBnv;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-play-state: paused;
  animation-play-state: paused;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  display: inherit;
}

<div
  className="c0"
/>
`;

exports[`Animate Fill Mode Should take fillMode prop 1`] = `
.c0 {
  -webkit-animation: hVlBnv;
  animation: hVlBnv;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-play-state: paused;
  animation-play-state: paused;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  display: inherit;
}

<div
  className="c0"
/>
`;

exports[`Animate Time Should have 1s default time 1`] = `
.c0 {
  -webkit-animation: hVlBnv;
  animation: hVlBnv;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-play-state: paused;
  animation-play-state: paused;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  display: inherit;
}

<div
  className="c0"
/>
`;

exports[`Animate Time Should take time prop 1`] = `
.c0 {
  -webkit-animation: hVlBnv;
  animation: hVlBnv;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-play-state: paused;
  animation-play-state: paused;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  display: inherit;
}

<div
  className="c0"
/>
`;

exports[`Animate Trigger Should have default trigger of false 1`] = `
.c0 {
  -webkit-animation: hVlBnv;
  animation: hVlBnv;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-play-state: paused;
  animation-play-state: paused;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  display: inherit;
}

<div
  className="c0"
/>
`;

exports[`Animate Trigger Should take trigger prop 1`] = `
.c0 {
  -webkit-animation: hVlBnv;
  animation: hVlBnv;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-play-state: running;
  animation-play-state: running;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  display: inherit;
}

<div
  className="c0"
/>
`;
